flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotInfoWithUtilization.java

```diff
@@ -18,26 +18,37 @@
 
 package org.apache.flink.runtime.jobmaster.slotpool;
 
+import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.runtime.clusterframework.types.AllocationID;
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
 import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
 import org.apache.flink.runtime.jobmaster.SlotInfo;
 import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
 
+import java.util.function.Function;
+
 /**
  * Container for {@link SlotInfo} and the task executors utilization (freeSlots /
  * totalOfferedSlots).
  */
 public final class SlotInfoWithUtilization implements SlotInfo {
     private final SlotInfo slotInfoDelegate;
-    private final double taskExecutorUtilization;
+    private final Function<ResourceID, Double> taskExecutorUtilizationLookup;
 
-    private SlotInfoWithUtilization(SlotInfo slotInfo, double taskExecutorUtilization) {
+    private SlotInfoWithUtilization(
+            SlotInfo slotInfo, Function<ResourceID, Double> taskExecutorUtilizationLookup) {
         this.slotInfoDelegate = slotInfo;
-        this.taskExecutorUtilization = taskExecutorUtilization;
+        this.taskExecutorUtilizationLookup = taskExecutorUtilizationLookup;
     }
 
+    @VisibleForTesting
     double getTaskExecutorUtilization() {
-        return taskExecutorUtilization;
+        return taskExecutorUtilizationLookup.apply(
+                slotInfoDelegate.getTaskManagerLocation().getResourceID());
+    }
```

Comment:
It seems that this method has still not been removed. Have you just forgotten it or do you have other concerns 🤔